tmt/steps/__init__.py

```diff
@@ -448,45 +596,43 @@ class BasePlugin(Phase, metaclass=PluginIndex):
     # except for provision (virtual) and report (display)
     how: str = 'shell'
 
-    # Common keys for all plugins of given step
-    _common_keys: List[str] = []
-
-    # Keys specific for given plugin
-    _keys: List[str] = []
-
     # List of all supported methods aggregated from all plugins of the same step.
     _supported_methods: List[Method] = []
 
+    _data_class: Type[StepData] = StepData
+    data: StepData
+
+    # TODO: do we need this list? Can whatever code is using it use _data_class directly?
+    # List of supported keys
+    # (used for import/export to/from attributes during load and save)
+    @property
+    def _keys(self) -> List[str]:
+        return list(self._data_class.keys())
+
     def __init__(
             self,
             step: Step,
             data: StepData,
             workdir: tmt.utils.WorkdirArgumentType = None) -> None:
         """ Store plugin name, data and parent step """
 
-        # Ensure that plugin data contains name
-        if 'name' not in data:
-            raise tmt.utils.GeneralError(
-                f"Missing 'name' in the {step} step config "
-                f"of the '{step.plan}' plan.")
-
         # Initialize plugin order
-        if 'order' not in data or data['order'] is None:
-            order = tmt.utils.DEFAULT_PLUGIN_ORDER
-        else:
-            try:
-                order = int(data['order'])
-            except ValueError:
-                raise tmt.utils.SpecificationError(
-                    f"Invalid order '{data['order']}' in {step} config "
-                    f"'{data['name']}'. Should be an integer.")
+        # if 'order' not in data or data['order'] is None:
+        #     order = tmt.utils.DEFAULT_PLUGIN_ORDER
+        # else:
+        #     try:
+        #         order = int(data['order'])
+        #     except ValueError:
+        #         raise tmt.utils.SpecificationError(
+        #             f"Invalid order '{data['order']}' in {step} config "
+        #             f"'{data['name']}'. Should be an integer.")
```

Comment:
Why is this left / commented out?

Reply:
Hmm, looks like a leftover from the development stage. This has been performed by `__init__()`, so I moved the code to the normalization stage and commented it out here, so I could return to it if needed. And it's not needed, IIUIC.

Reply:
Fixed in [8d096fd](https://github.com/teemtee/tmt/pull/1461/commits/8d096fd6dedfb1f4dab9cea49f75946075af5836)